ls1intum/Artemis
# [Enhancement] - Renaming of course statistics

### Checklist

### Motivation and Context


- Integrated some name changes to naming suggestions in the statistics of a course. (in students view) 
- Also, pie chart on the left side of the statistics was refering to maximum scores, where it should actually be reachable score (difference is that max score are all the exercises of the course combined, reachable score are only the maximum exercise points if the participated exercises combined)

### Description

- Integrated some name changes to naming suggestions in the statistics of a course. (in students view) 
- Name changes in the course-card in the course overview for students (see screenshots)
- Pie chart now refers to the reachable scores
- Below the pie chart there is a summary of the points (analogue to the summary next to every exerciseGroup)
- Added a client test which tests if scores and points are calculated correctly

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Overview
3. Check that it says "Your overall score" in the tile
4. Choose a course
5. Navigate to the Statistics tab
6. Make sure that the the reachable score on the left side is in the chart as well as mentioned below the chart and that it is the sum of the reachable scores of all exerciseGroups
7.  Check that the overall score on the left side is the sum of the overall scores of the right side
8. Make sure that the name changes like shown in the screenshots are correctly displayed and make sure there are translations also in german
9. Check that the tooltips contain the text as mentioned below under the screenshot
10. Check again that your points in the whole course (on the left side) is the same percentage as in the tile in the course overview page (see step 3.)


### Screenshots


Course overview (Your current score -> Your overall score):
![Screenshot 2021-01-12 at 17 22 13](https://user-images.githubusercontent.com/44401560/104464482-43cdcd80-55b3-11eb-9599-e35cef270a16.png)

1. This score should be the same score as your course points in the following picture

The course statistic changes:
![Screenshot 2021-01-13 at 14 51 43](https://user-images.githubusercontent.com/44401560/104463339-f13fe180-55b1-11eb-84ba-111bc044d0f4.png)


As well as the tooltips (see screenshot):
2. "Reachable points are the sum of the maximum points of all exercises of this type which have already been assessed, independent of your participation"
3. Overall points are the sum of the maximum points of all exercises of this type.
4. "Reachable points are the sum of the maximum points of all exercises which have already been assessed, independent of your participation"
5. "Overall points are the sum of the maximum points of all exercises"
